core/src/main/resources/scala-native/ffi.c

```diff
@@ -0,0 +1,7 @@
+#include <curl/curl.h>
```

Comment:
The CI seems to be failing because it cannot find `libcurl` library at linking. It seems, like the `CCurl.libcurlPlatformCompat.instance` was never referenced in projects not testing/using curl. To fix it we can introduce a compiletime guard using macro preprocessor definitions. For that we can wrap this file content in 
```c 
#if defined(STTP_CURL_FFI)
#include <curl/curl.h> 
...
#endif
``` 

and add `@define("STTP_CURL_FFI")` to symbols that use `@link("curl")` or `@link("libcurl")`. 
At this point this C file content would be compiled only when curl definitions would be referenced. See https://github.com/scala-native/scala-native/pull/3427
Unfortunately, it seems like `@define` annotation was not backported from main branch to 0.4.x, I'll make try to include it and release 0.4.18 this weekend.

Reply:
@WojciechMazur I'm updating this PR to scala-native 0.5.1 now, I guess this is still relevant?